### PR TITLE
version command for console

### PR DESCRIPTION
**Pull request by rendayigit**

### CODE REVIEW

File: Client_console/client.cpp [hunk 1]
```diff
@@ -1,13 +1,33 @@
 #include "client.h"
 
+#include <QtCore/qthread.h>
```
Comment (rendayigit) [inline Client_console/client.cpp:3]: incorrect include

File: Client_console/client.cpp [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
+
 #include "../common.h"
 #include "../path.h"
 
 using namespace std;
 
 Client::Client() : Operations(Paths().getClientCmdsYaml()) {}
 
-void Client::onReceived(QByteArray message) { std::cout << message.toStdString() << std::endl; }
+std::string serverVersion;
```
Comment (rendayigit) [inline Client_console/client.cpp:12]: keep this line in the header (there already seems to be the same line in the header so simply delete this i guess)

File: Client_console/client.cpp [hunk 3]
```diff
@@ -1,13 +1,33 @@
 #include "client.h"
 
+#include <QtCore/qthread.h>
+
 #include "../common.h"
 #include "../path.h"
 
 using namespace std;
```
Comment (rendayigit) [inline Client_console/client.cpp:8]: since we have this line we can remove all `std::`'s from this file

File: Client_console/client.cpp (hunk 1 above)
Comment (mrtkrkrt) [reply Client_console/client.cpp:3]: applied

File: Client_console/client.cpp (hunk 2 above)
Comment (mrtkrkrt) [reply Client_console/client.cpp:12]: applied

**Review by rendayigit — COMMENTED** [review]
image

**Comment by rendayigit** [comment]
image

**Comment by rendayigit** [comment]
image

**Comment by mrtkrkrt** [comment]
Inspect this branch.

**Comment by mrtkrkrt** [comment]
I think this issue is done completely. We can merge this branch to main branch.

File: Client_console/client.cpp (hunk 2 above)
Comment (rendayigit) [reply Client_console/client.cpp]: u didnt remove the line in the header, anyway ill remove it

File: Client_console/client.cpp [hunk 4]
```diff
@@ -1,15 +1,45 @@
...
+            QString::fromStdString(message.toStdString()).split(" = ").at(1).toFloat();
+
+        if (serverVersionNumeric < clientVersionNumeric) {
+            QString message = "WARNING Server version is older than the Client version !!!";
+            qWarning().noquote() << message;
+            Log().Warn(message);
+        } else if (serverVersionNumeric > clientVersionNumeric) {
+            QString message = "WARNING Client version is older than the Server version !!!";
+            qWarning().noquote() << message;
+            Log().Warn(message);
```
Comment (rendayigit) [inline Client_console/client.cpp:33]: Add same logic to gui

Comment (rendayigit) [reply Client_console/client.cpp:33]: Will be done at https://github.com/rendayigit/SpaceLoader/issues/120

**Pull request merged**